### PR TITLE
Update for new folder structure

### PR DESCRIPTION
**Pull request by JamesSample**
Based on the discussion [here](https://github.com/SeaBee-no/documentation/issues/23), we agreed that pilots should be able to name the flight folders however they wish, and that all required information should be included in the configuration file.

Main changes:

- `config.yaml` is now named `config.seabee.yaml` to avoid conflicts with other configuration files produced by e.g. NodeODM

- `flights_runner.ipynb` now recursively searches the `base_dir` for folders containing `config.seabee.yaml` and attempts to process these, instead of just looking for flight folders one level down from `base_dir`. This should make it easier to process mission data arranged in a more hierarchical way.

- The name of the flight folder is ignored completely. Instead, `grouping`, `area` and `datetime` are now mandatory in `config.seabee.yaml`.

- The `config.yaml` files previously used by Sindre have been updated to include the new mandatory attributes (based on the folder names) and the files renamed as `config.seabee.yaml`.

- `sb.ortho.list_images` now ignores case in the file extension i.e. `ext='jpg'` will find both `.jpg` and `.JPG` files, etc. (see #5)

Tested by reprocessing [this](https://geonode.seabee.sigma2.no/catalogue/#/dataset/667) dataset, and it seems to work OK. As long as I haven't completely messed it up, final testing is probably easiest once it's merged.

### CODE REVIEW

**Review by knl88 — APPROVED** [review]
Nice, lgtm👍

File: seabeepy/geo.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def upload_raster_to_geoserver(fpath, user, password, workspace="geonode"):
         workspace=workspace,
     )
 
-    return layer_name
+    return None
```
Comment (knl88) [inline seabeepy/geo.py:115]: we don't have to return anything to return None

File: seabeepy/ortho.py [hunk 2]
```diff
@@ -274,18 +307,14 @@ def check_file_count(dir_path, verbose=False):
...
-    config_path = os.path.join(dir_path, "config.yaml")
-    with open(config_path, "r") as stream:
-        data = yaml.safe_load(stream)
+    data = parse_config(dir_path)
     nfiles_expected = data["nfiles"]
-
     if nfiles_found == nfiles_expected:
         return True
     else:
         if verbose:
```
Comment (knl88) [inline seabeepy/ortho.py:315]: remove the else?

File: seabeepy/geo.py (hunk 1 above)
Comment (JamesSample) [reply seabeepy/geo.py:115]: Yeah, I know. I like to explicitly return `None` in cases where the return value is likely to be used/tested, but in this case I just lazily changed `layer_name` to `None`. I'll remove it

Comment (knl88) [reply seabeepy/geo.py:115]: :+1:  you will hate [Julia](https://groups.google.com/g/julia-users/c/4RVR8qQDrUg) :grinning:

Comment (JamesSample) [reply seabeepy/geo.py:115]: Yep, I already hate Julia! I'm hoping Mojo is what I've been waiting for...

File: seabeepy/ortho.py [hunk 3]
```diff
@@ -142,25 +148,38 @@ def check_subdir_exists(dir_path, subdir):
...
             "organisation": str,
             "mosaic": bool,
             "publish": bool,
-            "theme": lambda s: s in ("Seabirds", "Mammals", "Habitat"),
+            "theme": lambda s: s.lower() in ("seabirds", "mammals", "habitat"),
+            Optional("spectrum_type"): Or(
+                lambda s: s.lower() in ("rgb", "ms", "hsi"), None
+            ),
+            Optional("elevation"): Or(And(int, lambda x: x >= 0), None),
             Optional("creator_name"): Or(str, None),
```
[the remark this answers is not in the document]
Comment (JamesSample) [reply seabeepy/ortho.py:183]: Do you mean just setting

```python
ORTO_CONFIG= Schema(
        {
            "grouping": str,
            "area": str,
            "datetime": And(
                str,
                Or(
                    lambda date: dt.datetime.strptime(date, "%Y%m%d"),
                    lambda date: dt.datetime.strptime(date, "%Y%m%d%H%M"),
                ),
            ),
            "nfiles": And(int, lambda n: n > 0),
            "organisation": str,
            "mosaic": bool,
            "publish": bool,
            "theme": lambda s: s.lower() in ("seabirds", "mammals", "habitat"),
            Optional("spectrum_type"): Or(
                lambda s: s.lower() in ("rgb", "ms", "hsi"), None
            ),
            Optional("elevation"): Or(And(int, lambda x: x >= 0), None),
            Optional("creator_name"): Or(str, None),
            Optional("project"): Or(str, None),
            Optional("odm_options"): {
                Optional("dsm"): bool,
                Optional("dtm"): bool,
                Optional("cog"): bool,
                Optional("orthophoto-compression"): lambda s: s
                in ("JPEG", "LZW", "PACKBITS", "DEFLATE", "LZMA", "NONE"),
                Optional("orthophoto-resolution"): Or(int, float),
                Optional("dem-resolution"): Or(int, float),
                Optional("max-concurrency"): int,
                Optional("auto-boundary"): bool,
                Optional("use-3dmesh"): bool,
                Optional("fast-orthophoto"): bool,
                Optional("pc-rectify"): bool,
                Optional("split"): int,
                Optional("split-overlap"): int,
                Optional("crop"): And(Or(int, float), lambda x: x >= 0),
                Optional("pc-quality"): lambda s: s
                in ("ultra", "high", "medium", "low", "lowest"),
                Optional("feature-quality"): lambda s: s
                in ("ultra", "high", "medium", "low", "lowest"),
            },
        }
    )
```

at the top of file file, or actually moving it to a separate `constants.py`, or similar?

**Pull request merged**